mlir/lib/Dialect/GPU/IR/GPUDialect.cpp

```diff
@@ -634,7 +635,9 @@ void LaunchOp::build(OpBuilder &builder, OperationState &result,
                       getBlockSizeY, getBlockSizeZ});
   if (dynamicSharedMemorySize)
     result.addOperands(dynamicSharedMemorySize);
-
+  if (dynamicSharedMemorySizeAttr)
```

Comment:
This `build` method is allowing the creation of `gpu.launch` with both dynamic and constant valid sizes, from the rest of the patch it's clear that they are mutually exclusive. A check on the verifier should be added ensuring they are exclusive.

Reply:
Good point! I've added a verifier.